### DISCUSSION TITLE
Mix dependencies with `end` cut the deps list short

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#24428] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.93.7

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We were trying to use renovate to scan a large Elixir/Mix project. The scan was successful, but we noticed that the dep count that renovate was reporting was significantly smaller than what we had and that it was stopping short on a dep called `hammer`.

Right after `hammer`, there's a dep called `hammer_backend_redis`. I was inspecting the code that performs the extraction, and it looks like that it might be the case that the extraction is being cutting short because `hammer_backend_redis` contains `end`, as that seems to be the test to detect the `end` keyword that would match `defp`: https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/mix/extract.ts#L35

Minimal repro should be: https://github.com/tpresa-remote/renovate-hex-dep-with-end-minimal-repro

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by viceice** [comment]
looks like a bug, but the repo isn't minimal. I think you should be able to reproduce with only two deps.

you can also remove all auxillary files, because we don't need them for extraction

**Reply by ghost** [reply]
I changed the repo to a minimal one, it is under the same URL.

**Comment by rarkins** [comment, marked as answer]
Fix PR: #24665
